packages/core/src/supported-chains/index.ts

```
@@ -0,0 +1,86 @@
+ * 
+ * @example
+ * const selector = getChainSelectorByName("ethereum");
+ * console.log(selector); // Outputs: 1
+ */
+export function getChainSelectorByName(name: string): number {
+    const lowerCaseName = name.toLowerCase();
+
+    for (const [selector, { names }] of Object.entries(chainSelectors)) {
+        if (names.includes(lowerCaseName)) {
+            return Number(selector);
+        }
+    }
+
+    throw new Error(`Invalid or unsupported Chain Name: ${name}`);
+}
+
+/**
+ * @title getAllSupportedChains
+ * @description Retrieves all supported chains with their respective chain selector, name, and image path.
+ * 
+ * @returns {Array<{ chainSelector: number, name: string, image: string }>}
+ * - An array of objects, each containing the chain selector, name, and image path for a supported chain.
+ * 
+ * @example
+ * const chains = getAllSupportedChains();
+ * console.log(chains);
+ * // [
+ * //   { chainSelector: 0, name: "equito", image: "images/equito.svg" },
+ * //   { chainSelector: 1, name: "ethereum", image: "images/ethereum.svg" },
+ * //   ...
+ * // ]
+ */
+export function getAllSupportedChains(): Array<{ chainSelector: number, name: string, image: string }> {
+    const supportedChains = [];
+
+    for (const [selector, { names, image }] of Object.entries(chainSelectors)) {
+        if (names[0]) {
+            supportedChains.push({
+                chainSelector: Number(selector),
+                name: names[0],
+                image,
+            });
+        }
+    }
+
+    return supportedChains;
+}
+
+/**
+ * @description Get the primary name of the chain by its selector.
+ * @param {number} chainSelector - The selector value for the chain.
+ * @returns {string} - The primary name of the chain.
+ * @throws {Error} - Throws an error if the chain selector is not found.
+ */
+export function getNameByChainSelector(chainSelector: number): string | undefined {
```

```suggestion
export function getNameByChainSelector(chainSelector: number): string {
```

let's throw an error if doesn't exists